### PR TITLE
Andrew submission

### PR DESCRIPTION
**Pull request by freblisna**
Hi Yael,

Here is my submission for the coding exercise you shared. 

1) I implemented an "n_convLayers" class that can generates a flexible number of convolutions per level of the UNET. This class is then called by the "Backbone" class during a loop that creates the appropriate number of UNET levels. The parameters you requested that the user be able to specify are all included in the class as well as its documentation. Since this was my first time using PyTorch, I opted not to parameterize the optional settings and I went with only ReLU activation as well as max-pool for the contracting side & convTran2D+interpolation for the expanding side.

2) For training, I implemented a function that takes the trainset and evalset as mandatory inputs and then various other settings that apply to the model architecture or training. This function both generates the UNET and trains it, returning the trained model and printing various outputs along the way. It can also output graphs of the training and evaluation loss during training for the user to monitor. I also implemented a simple version of early-stopping since I noticed during development that the training loss tended to plateau, as did the validation loss. It follows the logic: "If the difference between the most recent validation loss (as measured after n batches of training in an epoch) and the previous 4 validation loss measurements (as measured similarly at those previous moments in training) is less than some threshold that can be set by the user, than the epoch should end early". Similarly, "If the difference between the most recent final validation loss for an epoch and that of the previous 4 epochs is less than some other user specified threshold, then the training overall should end early. I also included an output function to show the final "moved image" results, to confirm that the model being applied to the test set indeed warps the images in the expected way.

Let me know what you think, and any kind of feedback on this would be very helpful for me!

### CODE REVIEW

**Comment by freblisna** [comment]
Just saw it had two commits in here. The more recent / 2nd one is the final version. The only difference is I added some more comments to the code for the second one.
